src/core.js

```diff
@@ -848,7 +848,7 @@ $.extend( $.validator, {
 			// aria-describedby should directly reference the error element
 			if ( describer ) {
 				selector = selector + ", #" + describer
-					.replace( /'/g, "\\'" )
+					.replace( /('|:|\.|\[|\]|\$)/g, "\\$1" )
```

Comment:
Line 814 is in `showLabel` method, and line 851 is in `errorFor` method. So what do you suggest to solve this ? a global variable or something else ?